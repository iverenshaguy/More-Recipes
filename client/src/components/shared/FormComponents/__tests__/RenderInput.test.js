import React from 'react';
import { RenderInput } from '../index';

describe('Form Components: RenderInput', () => {
  const cleanProps = {
    input: { name: 'email' },
    placeholder: 'jane',
    type: 'email',
    meta: {
      touched: false,
      error: null,
      asyncValidating: false
    }
  };

  const dirtyProps = {
    input: { name: 'email' },
    placeholder: 'jane',
    type: 'email',
    meta: {
      touched: true,
      error: 'This is an error',
      asyncValidating: false
    }
  };

  it('renders correctly', () => {
    const mountedWrapper = mount(<RenderInput {...cleanProps} />);
    expect(toJson(mountedWrapper)).toMatchSnapshot();
    expect(mountedWrapper.find('div.invalid-feedback').exists()).toBeFalsy();
    expect(mountedWrapper.find('input').hasClass('is-invalid')).toBeFalsy();
    mountedWrapper.unmount();
  });

  it('shows error when found', () => {
    const mountedWrapper = mount(<RenderInput {...dirtyProps} />);

    expect(toJson(mountedWrapper)).toMatchSnapshot();
    expect(mountedWrapper.find('.invalid-feedback').text()).toEqual('This is an error');
    expect(mountedWrapper.find('input').hasClass('is-invalid')).toBeTruthy();
    mountedWrapper.unmount();
  });

  it('shows \'Checking\' when async validating', () => {
    const mountedWrapper = mount(<RenderInput
      {...cleanProps}
      meta={{ asyncValidating: true, touched: true }}
    />);

    expect(toJson(mountedWrapper)).toMatchSnapshot();
    expect(mountedWrapper.find('small').hasClass('form-text')).toBeTruthy();
    expect(mountedWrapper.find('small').text()).toEqual('Checking...');
    mountedWrapper.unmount();
  });
});
