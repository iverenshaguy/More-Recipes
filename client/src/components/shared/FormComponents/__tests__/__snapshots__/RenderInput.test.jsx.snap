// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Components: RenderInput renders correctly 1`] = `
<RenderInput
  handleBlur={[Function]}
  handleChange={[Function]}
  handleFocus={[Function]}
  meta={
    Object {
      "asyncValidating": false,
      "error": null,
      "touched": false,
    }
  }
  name="email"
  placeholder="jane"
  type="email"
  value=""
>
  <Input
    className=""
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="jane"
    tag="p"
    type="email"
    value=""
  >
    <input
      className="form-control"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="jane"
      type="email"
      value=""
    />
  </Input>
</RenderInput>
`;

exports[`Form Components: RenderInput shows 'Checking' when async validating 1`] = `
<RenderInput
  handleBlur={[Function]}
  handleChange={[Function]}
  handleFocus={[Function]}
  meta={
    Object {
      "asyncValidating": true,
      "touched": true,
    }
  }
  name="email"
  placeholder="jane"
  type="email"
  value=""
>
  <Input
    className="is-valid"
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="jane"
    tag="p"
    type="email"
    value=""
  >
    <input
      className="is-valid form-control"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="jane"
      type="email"
      value=""
    />
  </Input>
  <small
    className="form-text"
  >
    Checking...
  </small>
</RenderInput>
`;

exports[`Form Components: RenderInput shows error when found 1`] = `
<RenderInput
  handleBlur={[Function]}
  handleChange={[Function]}
  handleFocus={[Function]}
  meta={
    Object {
      "asyncValidating": false,
      "error": "This is an error",
      "touched": true,
    }
  }
  name="email"
  placeholder="jane"
  type="email"
  value="jane@smith.me"
>
  <Input
    className="is-invalid"
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="jane"
    tag="p"
    type="email"
    value="jane@smith.me"
  >
    <input
      className="is-invalid form-control"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="jane"
      type="email"
      value="jane@smith.me"
    />
  </Input>
  <div
    className="invalid-feedback"
  >
    This is an error
  </div>
</RenderInput>
`;
